ion-schema/src/violation.rs

```diff
@@ -146,3 +153,59 @@ impl fmt::Display for ViolationCode {
         )
     }
 }
+
+#[cfg(test)]
+mod violation_tests {
+    use crate::ion_path::IonPath;
+    use crate::violation::{Violation, ViolationCode};
+
+    #[test]
+    fn violation_equivalence() {
```

Comment:
Can you add another test case with multiple levels of nesting?

Reply:
Sure. I will add that.

Reply:
`PartialEq` had missing `ion_path` equivalence check. Added a condition check for that along with more tests.